### ISSUE TITLE
Allow specifying scopes using refresh_token grant

### ISSUE THREAD

**Issue by sburst** [IdentityServer/IdentityServer4#3644]
**Is your feature request related to a problem? Please describe.**
I have a portal with various applications, and each one needs a token with a different scope.
At the moment, using the refresh token, I'm providing a token to each app, but all tokens have all scopes originally requested.

**Describe the solution you'd like**
I would like to specify the scope when using the refresh_token grant.

The scope list should be a subset of originally requested scopes.

For example:
```
POST /connect/token HTTP/1.1
Content-Type: application/x-www-form-urlencoded

client_id=client&client_secret=password&grant_type=refresh_token&scope=SCOPE_SUBSET&refresh_token=1a2b3c4d5e6f...7g8h9i
```

and the returned token should contain only the specified scope subset.

**Comment by leastprivilege** [comment]
This is currently not implemented. Will add it as a feature request for 4.0. thanks!

**Comment by JuliusPC** [comment]
I have a similar problem to solve and was surprised that IdentityServer didn’t worked as expected, since [this behavior is defined in the OAuth 2.0 spec](https://tools.ietf.org/html/rfc6749#section-6). In my opinion, you could classify the behavior as a bug and not a missing feature.

It would be great if this makes it into IdentityServer 5.

**Comment by leastprivilege** [comment]
well - it is OPTIONAL.

But agreed that this would be nice.

**Comment by JuliusPC** [comment]
> well - it is OPTIONAL.

I should have consulted [RFC 2119](https://tools.ietf.org/html/rfc2119) before thinking of OPTIONAL as optional parameter instead optional implementation...

**Comment by Brunni** [comment]
I was also suprised, that the "Scope" parameter on the Token-Endpoint just does nothing and is completely ignored.

I think for some services you might want to send only a less powerful AccessToken  (e.g. less user claims, less scopes).

Increasing the number scopes via Token-Endpoint might not be possible, right?

**Comment by JuliusPC** [comment]
> I was also suprised, that the "Scope" parameter on the Token-Endpoint just does nothing and is completely ignored.

I tested the PHP library [OAuth 2.0 Server](https://oauth2.thephpleague.com/) a few days ago. I found that this library considered the scope parameter as expected. This behavior is therefore not an exaggeratedly fussy interpretation of the spec.

> Increasing the number scopes via Token-Endpoint might not be possible, right?

No. [The spec states](https://tools.ietf.org/html/rfc6749#section-6): _„The requested scope MUST NOT include any scope not originally granted by the resource owner“_
